### PR TITLE
Code changes to extract short dump metrics

### PR DESCRIPTION
**Pull request by hsridharan**
RFC Call is made to the SAP methods: 'SDF/GET_DUMP_LOG' by passing the following inputs : 
1.	Date_From
2.	Time_From
3.	Date_To
4.	Time_to
We get two tables as result from SAP: ES_E2E_LOG_STRUC_DESC and	ET_E2E_LOG which is combined as one JSON and pushed to Log analytics. (Screenshot below from Log Analytics)
image

Addition Library - **pandas** is included in the docker file which is used to create a data frame copying some column names from 
ES_E2E_LOG_STRUC_DESC  to ET_E2E_LOG table and form a single table

Testing: Input parameter was updated according to the data available in SAP and verified that we get the output back and its pushed to LA successfully.

### CODE REVIEW

**Comment by PakDLiu** [comment]
Can you add a description?
How it works, the library you used, testing you did, sample data, etc

**Comment by hsridharan** [comment]
@PakDLiu - Sorry I totally missed updating the same. Can you please check the description now

**Comment by PakDLiu** [comment]
Looks like the data contains the user that executed the process.
This would be considered PII if I am not mistaken. If so, we can not store this in our Kusto clusters.
I am not sure if we can even have this information pass through our system. Is there a way we can exclude that information from the API calls?

Also, can you update [this document](https://microsoft.sharepoint.com/:x:/r/teams/SAPonAzure516/_layouts/15/Doc.aspx?sourcedoc=%7B87169ce7-3635-4ebf-b905-08e495ab5172%7D&action=editnew&cid=0FF21311-D322-476A-9CC2-08F5F1A27FE1&wdLOR=cA72040C2-2922-4E49-9B19-789CBE85EA03) with the new metrics?

**Comment by PakDLiu** [comment]
Question: Would the `Error_Short_Text_s` field also contain PII? I am not sure...

**Comment by hsridharan** [comment]
@PakDLiu - No its not PII. It contains text short dump error messages. E.g -" BASIC_CO" cannot be interpreted as a number, No more memory available to add rows to an internal table. etc.

File: sapmon/payload/provider/sapnetweaver.py [hunk 1]
```diff
@@ -938,7 +942,56 @@ def _actionGetSwncWorkloadMetrics(self) -> None:
...
+            # only update state on successful query attempt
+            self.updateState()
+
+        except Exception as e:
+            self.tracer.error("%s exception trying to fetch short dumps workload metrics for %s [%d ms], error: %s",
+                              self.logTag, 
+                              sapHostnameStr,
+                              TimeUtils.getElapsedMilliseconds(latencyStartTime),
+                              e,
+                              exc_info=True)
```
Comment (MikeFrei) [inline sapmon/payload/provider/sapnetweaver.py:992]: Harsha I just realized, you are pulling in some of Al's changes here around logging right?

Comment (MikeFrei) [reply sapmon/payload/provider/sapnetweaver.py:992]: I think this is probably a no-op, i just wanted to make sure there were no other related changes for this

Comment (hsridharan) [reply sapmon/payload/provider/sapnetweaver.py:992]: No Mike. Do we have a separate branch that I need to pull into Mine? I just updated  exc_info=True in all the exceptions which are included as a part of the short dump codebase.

File: sapmon/payload/netweaver/metricclientfactory.py [hunk 2]
```diff
@@ -76,8 +81,8 @@ def getMetricClient(tracer: logging.Logger,
                                    serverTimeZone=kwargs.get("serverTimeZone", None))
 
         except ImportError as importEx:
-            tracer.error("failed to import pyrfc module, unable to initialize NetWeaverRfcClient: ", importEx)
+            tracer.error("failed to import pyrfc module, unable to initialize NetWeaverRfcClient: ", importEx, exc_info=True)
```
Comment (MikeFrei) [inline sapmon/payload/netweaver/metricclientfactory.py:84]: the more I think about this the more I think we might want to remove this exc_info=True argument from all your logging calls until Al has had a chance to validate that this did not impact his queue-based operational log statements.  There is a chance that adding this argument is causing the custom queue logging implementation to fail.  Until we confirm it is not an issue, I think it is a risk to include.

**Comment by PakDLiu** [comment]
I saw you set the customer analytics to false, which is good.
I am not sure if we can have PII flow through the Collector VM (especially when hybrid is complete).
Is this field required? Is there a way we can exclude this field in the query level?

File: sapmon/payload/provider/sapnetweaver.py (hunk 1 above)
Comment (MikeFrei) [reply sapmon/payload/provider/sapnetweaver.py:992]: ah okay in that case let's just remove all exc_info=True flags in your logging calls for now until Al has had a chance to validate these are not causing issues with queue based logging.

File: sapmon/payload/netweaver/metricclientfactory.py (hunk 2 above)
Comment (hsridharan) [reply sapmon/payload/netweaver/metricclientfactory.py]: @MikeFrei - Removed that from all the related files.

File: sapmon/payload/provider/sapnetweaver.py (hunk 1 above)
Comment (hsridharan) [reply sapmon/payload/provider/sapnetweaver.py]: Done removed.

**Review by PakDLiu — APPROVED** [review]

**Review by persiaAziz — APPROVED** [review]

**Pull request merged**